src/backend/distributed/utils/maintenanced.c

```diff
@@ -305,7 +305,7 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 	pgstat_report_appname("Citus Maintenance Daemon");
 
 	/* enter main loop */
-	for (;;)
+	while (!got_SIGTERM)
```

Comment:
This change definitely makes sense.

I do wonder about the overall behaviour in case of StopMaintenanceDaemon. We remove dbData from the hash, but might still be in the middle of an iteration in the main loop and we still access `myDbData` in the `if (rc & WL_LATCH_SET)` section. Could that crash?

Reply:
Yes, the hope is the mainloop exits before the hashtable has an insert, since HASH_REMOVE itself leaves the entry, but ideally we would be locking each loop iteration & making sure myDbData isn't stale